internal/backend/sessions/sessionworkflows/workflow.go

```diff
@@ -389,19 +390,37 @@ func (w *sessionWorkflow) run(wctx workflow.Context) (prints []string, err error
 	cbs := sdkservices.RunCallbacks{
 		NewRunID: newRunID,
 		Load:     w.load,
-		Call: func(callCtx context.Context, rid sdktypes.RunID, v sdktypes.Value, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error) {
-			if callCtx.Value(workflowContextKey) == nil {
+		Call: func(callCtx context.Context, runID sdktypes.RunID, v sdktypes.Value, args []sdktypes.Value, kwargs map[string]sdktypes.Value) (sdktypes.Value, error) {
+			if xid := v.GetFunction().ExecutorID(); xid.ToRunID() == runID && w.executors.GetCaller(xid) == nil {
+				// This happens only when we are during initial evaluation (the first run because invoking the entrypoint function),
+				// and the runtimes tries to call itself in order to start an activity with its own functions.
```

Comment:
```suggestion
				// This happens only during initial evaluation (the first run because invoking the entrypoint function),
				// and the runtime tries to call itself in order to start an activity with its own functions.
```

Reply:
fixed